nautobot/extras/fields.py

```diff
@@ -0,0 +1,96 @@
+from functools import partialmethod
+
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.utils.encoding import force_str
+from django.utils.hashable import make_hashable
+
+from nautobot.utilities.forms import DynamicModelChoiceField
+
+
+class LimitedChoiceField(models.ForeignKey):
+    """
+    Abstract model database field that automatically limits custom choices.
+
+    The limit_choices_to for the field are automatically derived from:
+
+        - the content-type to which the field is attached (e.g. `dcim.device`)
```

Comment:
I don't have exact wording in mind but think it would be good to elaborate on this statement a bit for clarity. This is a bit complex/confusing of a feature.